server/website/website/utils.py

```diff
@@ -568,12 +568,14 @@ def check_and_run_celery():
     if 'pong' in celery_status:
         return 'celery is running'
 
-    rabbitmq_status = os.popen('telnet localhost 5672').read()
+    rabbitmq_url = common.BROKER_URL.split('@')[-1]
+    hostname = rabbitmq_url.split(':')[0]
+    port = rabbitmq_url.split(':')[1].split('/')[0]
+    rabbitmq_status = os.popen('telnet {} {}'.format(hostname, port)).read()
     if 'Connected' in rabbitmq_status:
         LOG.info('Rabbitmq is running.')
     else:
         LOG.warning('Rabbitmq is not running.')
-        return 'Rabbitmq is not running.'
```

Comment:
You should still return this error message here if RabbitMQ is not running. The attempts to restart Celery afterward will never succeed if RabbitMQ is down and we end up returning the wrong error message.